backend/bitboxbase/mdns/manager.go

```diff
@@ -0,0 +1,183 @@
+			bitboxBase.Close()
+			delete(manager.baseDeviceInterface, bitboxBaseID)
+			manager.onUnregister(bitboxBaseID)
+			manager.log.WithField("bitboxbase-id", bitboxBaseID).Info("Unregistered bitboxbase")
+		}
+	}
+
+	// Make the id the address for now, later the pairing should be factored in here as well.
+	bitboxBaseID := address
+
+	// Skip if already registered.
+	if _, ok := manager.baseDeviceInterface[bitboxBaseID]; ok {
+		return false, errp.New("Base already registered")
+	}
+
+	manager.log.WithField("host", manager.getDetectedBases()[address]).WithField("address", address)
+	baseDevice, err := bitboxbase.NewBitBoxBase(address, bitboxBaseID, manager.config, manager.bitboxBaseConfigDir)
+
+	if err != nil {
+		manager.log.WithError(err).Error("Failed to register Base")
+		return false, err
+	}
+	manager.baseDeviceInterface[bitboxBaseID] = baseDevice
+	if err := manager.onRegister(baseDevice); err != nil {
+		manager.log.WithError(err).Error("Failed to execute on-register")
+		return false, err
+	}
+	return true, nil
+}
+
+// RemoveBase allows external objects to delete a manager entry.
+func (manager *Manager) RemoveBase(bitboxBaseID string) {
+	if _, ok := manager.baseDeviceInterface[bitboxBaseID]; ok {
+		delete(manager.baseDeviceInterface, bitboxBaseID)
+		manager.removedIDs[bitboxBaseID] = true
+	}
+}
+
+// checkIfRemoved returns true if a bitboxbase was detected in, but is not reachable anymore.
+func (manager *Manager) checkIfRemoved(bitboxBaseID string) bool {
+	// TODO implement this function, should check if the bitboxBase is reachable, if not properly unregister it.
+	return false
+}
+
+// mdnsScan scans the local network forever for BitBox Base devices.
+func (manager *Manager) mdnsScan() {
+	for {
+		entries := make(chan *micromdns.ServiceEntry)
+		go func() {
+			for entry := range entries {
+				host := entry.Host
+				resolvedHosts, err := net.LookupHost(host)
+				if err != nil {
+					manager.log.WithError(err).Error("Failed to resolve hostname: ", host)
+					continue
+				}
+				baseIPv4 := resolvedHosts[0] + ":" + strconv.Itoa(entry.Port)
+
+				deviceInfo := BaseDeviceInfo{
+					Hostname: host,
+					IPv4:     baseIPv4,
+				}
+
+				// Inform the backend that a new Base has been detected
+				// Even though micromdns.Lookup is supposed to filter based on mDNS service, sometimes it
+				// discovers other devices so we also filter them out here
+				if _, ok := manager.getDetectedBases()[deviceInfo.Hostname]; !ok && strings.Contains(entry.Name, service) {
+					manager.onDetect(deviceInfo)
+				}
+			}
+		}()
+
+		ctx, cancel := context.WithTimeout(context.Background(), discoveryDuration)
+
+		// Start mdns lookup
+		err := micromdns.Lookup(service, entries)
+		if err != nil {
+			manager.log.WithError(err).Error("mDNS lookup failed for service: ", service)
+		}
+
+		<-ctx.Done()
+
+		// Wait one second for go routine shutdown, necessary to correctly filter by "_bitboxbase._tcp"
+		time.Sleep(time.Second * 1)
```

Comment:
timers can be fragile. Look into sync.WaitGroup (can do wg.Wait() do wait for goroutines to end).